### PR TITLE
fix: Helm app proxy chart auto fix

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

Helm app proxy chart git ops operation auto fix while updating and Git clone fix

https://docs.google.com/document/d/15dktRsXFZkofiV1LRRf-gV3F6QRvQt93ikd4i2Fyj_Q/edit

Fixes https://github.com/devtron-labs/devtron/issues/2752

## How Has This Been Tested?
* [x] Helm app create for already created git repo. 
* [x]  Helm app update if no proxy chart available, auto fixed this case. (if master branch is not available)

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/appStore/deployment/service/AppStoreDeploymentService.go [hunk 1]
```diff
@@ -995,7 +996,14 @@ func (impl AppStoreDeploymentServiceImpl) UpdateInstalledApp(ctx context.Context
 				err = impl.appStoreDeploymentArgoCdService.UpdateRequirementDependencies(environment, installedAppVersion, installAppVersionRequest, appStoreAppVersion)
 				if err != nil {
 					impl.logger.Errorw("error while commit required dependencies to git", "error", err)
-					return nil, err
+					statusError, ok := err.(*github.ErrorResponse)
```
Comment (kartik-579) [inline pkg/appStore/deployment/service/AppStoreDeploymentService.go:999]: check errors for all gitOps providers

File: pkg/appStore/deployment/service/AppStoreDeploymentService.go [hunk 2]
```diff
@@ -995,7 +995,13 @@ func (impl AppStoreDeploymentServiceImpl) UpdateInstalledApp(ctx context.Context
 				err = impl.appStoreDeploymentArgoCdService.UpdateRequirementDependencies(environment, installedAppVersion, installAppVersionRequest, appStoreAppVersion)
 				if err != nil {
 					impl.logger.Errorw("error while commit required dependencies to git", "error", err)
-					return nil, err
+					noTargetFound, _ := impl.appStoreDeploymentArgoCdService.ParseGitRepoErrorResponse(err)
+					if noTargetFound {
+						//if by mistake no content found while updating git repo, do auto fix
+						installAppVersionRequest, err = impl.appStoreDeploymentArgoCdService.OnUpdateRepoInInstalledApp(ctx, installAppVersionRequest)
```
Comment (kartik-579) [inline pkg/appStore/deployment/service/AppStoreDeploymentService.go:1001]: please handle error

File: pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go [hunk 3]
```diff
@@ -385,7 +389,13 @@ func (impl AppStoreDeploymentArgoCdServiceImpl) UpdateInstalledApp(ctx context.C
 	installAppVersionRequest, err := impl.updateValuesYaml(environment, installedAppVersion, installAppVersionRequest)
 	if err != nil {
 		impl.Logger.Errorw("error while commit values to git", "error", err)
-		return nil, err
+		noTargetFound, _ := impl.ParseGitRepoErrorResponse(err)
+		if noTargetFound {
+			//if by mistake no content found while updating git repo, do auto fix
+			installAppVersionRequest, err = impl.OnUpdateRepoInInstalledApp(ctx, installAppVersionRequest)
```
Comment (kartik-579) [inline pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go:395]: please handle error

File: pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go [hunk 4]
```diff
@@ -463,3 +473,27 @@ func (impl AppStoreDeploymentArgoCdServiceImpl) updateValuesYaml(environment *cl
...
+
+func (impl AppStoreDeploymentArgoCdServiceImpl) ParseGitRepoErrorResponse(err error) (bool, error) {
+	//update values yaml in chart
+	noTargetFound := false
+	if err != nil {
+		if errorResponse, ok := err.(*github.ErrorResponse); ok && errorResponse.Response.StatusCode == http.StatusNotFound {
+			impl.Logger.Errorw("no content found while updating git repo on github, do auto fix", "error", err)
+			noTargetFound = true
+		}
+		if errorResponse, ok := err.(azuredevops.WrappedError); ok && *errorResponse.StatusCode == http.StatusNotFound {
```
Comment (kartik-579) [inline pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go:485]: azure sometimes also sends pointe to this wrapped error. please handle that case too.

**Review by kartik-579 — CHANGES_REQUESTED** [review]

File: pkg/appStore/deployment/service/AppStoreDeploymentService.go (hunk 2 above)
Comment (vikramdevtron) [reply pkg/appStore/deployment/service/AppStoreDeploymentService.go:1001]: already handled, not required

File: pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go (hunk 3 above)
Comment (vikramdevtron) [reply pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go:395]: already handled, not required

**Review by kartik-579 — DISMISSED** [review]

**Review by kartik-579 — APPROVED** [review]

**Pull request merged**